### DISCUSSION TITLE
[A/L] (2021/22) Induzione - Esercizio 18

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#318] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152323213-27038723-6aab-4cb7-91e8-55155b482af2.png)

**Comment by Exyss** [comment]
L'errore sta nell'auto-confutazione da parte del passo induttivo:

- Nel caso base viene dimostrato che _"max(n, m) = 0, ovviamente n = m = 0"_. Viene, involontariamente, assunto dunque che _n = m = k_.
- Nel passo induttivo viene assunto che **_max(n, m) = k+1 e dunque che max(n-1, m-1) = k_**. Tuttavia, nel caso base stesso viene assunto che affinché max(n, m) = k è necessario che n = m = k, mentre nel passo induttivo abbiamo **n = m =/= k.**

L'assunzione viene quindi è falsa, rendendo la tesi invalida

----

Un'ulteriore auto-confutazione può essere trovata sempre all'interno del passo induttivo, questa volta andando a verificare che il processo induttivo funzioni anche per k = 0.

Nel passo induttivo viene detto che **_max(n, m) = k+1 e dunque che max(n-1, m-1) = k_**.
Tuttavia, ricordando che per k = 0 nel caso base viene dimostrato che max(n, m) = 0 se n = m = 0, questa volta ponendo k = 0 otteniamo max(n, m) = 1 e dunque che max(n-1, m-1) = 0 e, poiché sappiamo che per k = 0 abbiamo n = m = 0, otteniamo max(-1, -1) = 0.

Il passo induttivo falsifica il caso base stesso, dunque la tesi è falsa.

**Comment by LuginiAndrea** [comment]
Reputo entrambe le confutazioni di @Exyss errate.
Nella prima:

1. Non solo nel caso base, ma in tutti i passi viene assunto che $n = m = k$, in quanto per come funziona il $\max$, $k = m \lor k = n$, il che unito al fatto che $n = m$ implica $n = m = k$, ma su tutti i passi. L'errore nel ragionamento sta nel fatto che non viene mai assunto nel Passo induttivo che $n = m = k$, bensì viene assunto che $n = m = k+1$.
2. Anche qui, $n,m$ non sono valori fissati. Il testo ci dice che $\max\left(n-1, m-1\right) = 0$, ma $n \land m$ non sono gli stessi valori del caso baso, non capisco quindi come si possa dire che $n = m = 0$.

Procedendo alla confutazione dell'Induzione dell'esercizio:
Il problema sta nel fatto che $n-1$ ed $m-1$ non sono necessariamente $\in \mathbf{N}$.
Infatti poniamo $k = 0$. Ne segue che $\max\left(n, m\right) = 1 \implies n = 1 \lor m = 1$, ma non che $n = m = 1$, in quanto $n = 1 \land m = 0$ è comunque un assegnamento accettabile. 
Ne segue che $n-1 \in \mathbf{N} \land m - 1 \not\in \mathbf{N}$, il che rende non applicabile l'ipotesi induttiva

**Comment by Exyss** [comment]
A distanza di un anno affermo che la mia precedente dimostrazione (ormai nascosta ma ancora visibile per chi volesse un esempio di come ___non___ fare una dimostrazione ) è __assolutamente sbagliata__ in quanto vengono fatte assunzioni completamente distaccate dalla dimostrazione stessa. Difatti, come sottolineato da @LuginiAndrea, viene assunto che $m$ ed $n$ utilizzate nel passo induttivo siano le stesse del caso base, ovviamente ciò è errato.

---------

La dimostrazione di @LuginiAndrea è corretta, l'errore è nell'inapplicabilità dell'ipotesi inuttiva nel caso in cui $n = 0 \lor m = 0$ come spiegato
